app/controllers/relationships_controller.rb

```diff
@@ -0,0 +1,19 @@
+class RelationshipsController < ApplicationController
+  before_action :require_user
+
+  def index
+    @relationships = current_user.following_relationships
+  end
+
+  def create
+    leader = User.find(params[:leader_id])
+    Relationship.create(leader_id: params[:leader_id], follower: current_user) if current_user.can_follow?(leader)
```

Comment:
You can also do:

```ruby
Relationship.create(leader: leader, follower: current_user) if current_user.can_follow?(leader)
```

since you have the leader object.